File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let valToIndex = new Map()
    
    for (let i = 0; i < nums.length; i++) {
        const curVal = nums[i]
        if (valToIndex.has(target - curVal)) return [i, valToIndex.get(target - curVal)]
        valToIndex.set(curVal, i)
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash_table = {}
        for i, x in enumerate(nums):
            hash_table[x] = i
        
        for i, x in enumerate(nums):
            y = target - x
            if y in hash_table:
                j = hash_table[y]
                if i != j:
                    return [i, j]
# adding comment to test leethub to test dfsf